Bottom: d0acb4cfa438526b28dad0518929670c673867bd
Top:    d8ef89041aff262ed7e41824f9083bbe12047254
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-20 12:43:04 +0100

Refresh of hid-gtk-add-consolidated-api-f

---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 2500fbc..8f69ab4 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1441,10 +1441,16 @@ ghid_zoom_view_fit (void)
 void
 ghid_flip_view (Coord center_x, Coord center_y, bool flip_x, bool flip_y)
 {
+  int widget_x, widget_y;
+
+  /* Work out where on the screen the flip point is */
+  ghid_pcb_to_event_coords (center_x, center_y, &widget_x, &widget_y);
+
   ghid_flip_x = ghid_flip_x != flip_x;
-  ghid_flip_y = ghid_flip_x != flip_y;
+  ghid_flip_y = ghid_flip_y != flip_y;
 
-  /* XXX: PAN THE BOARD SO THE CENTER LOCATION REMAINS IN THE SAME PLACE */
+  /* Pan the board so the center location remains in the same place */
+  ghid_pan_view_abs (center_x, center_y, widget_x, widget_y);
 
   ghid_invalidate_all ();
 }
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 5c43057..7840b4d 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1406,10 +1406,16 @@ ghid_zoom_view_fit (void)
 void
 ghid_flip_view (Coord center_x, Coord center_y, bool flip_x, bool flip_y)
 {
+  int widget_x, widget_y;
+
+  /* Work out where on the screen the flip point is */
+  ghid_pcb_to_event_coords (center_x, center_y, &widget_x, &widget_y);
+
   ghid_flip_x = ghid_flip_x != flip_x;
   ghid_flip_y = ghid_flip_y != flip_y;
 
-  /* XXX: PAN THE BOARD SO THE CENTER LOCATION REMAINS IN THE SAME PLACE */
+  /* Pan the board so the center location remains in the same place */
+  ghid_pan_view_abs (center_x, center_y, widget_x, widget_y);
 
   ghid_invalidate_all ();
 }
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index aff8dcc..671915d 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -1266,21 +1266,20 @@ SwapSides (int argc, char **argv, Coord x, Coord y)
   int comp_showing =   LAYER_PTR (PCB->LayerGroups.Entries[comp_group][0])->On;
   int solder_showing = LAYER_PTR (PCB->LayerGroups.Entries[solder_group][0])->On;
 
-
   if (argc > 0)
     {
       switch (argv[0][0]) {
         case 'h':
         case 'H':
-          ghid_flip_view (x, y, true, false);
+          ghid_flip_view (gport->pcb_x, gport->pcb_y, true, false);
           break;
         case 'v':
         case 'V':
-          ghid_flip_view (x, y, false, true);
+          ghid_flip_view (gport->pcb_x, gport->pcb_y, false, true);
           break;
         case 'r':
         case 'R':
-          ghid_flip_view (x, y, true, true);
+          ghid_flip_view (gport->pcb_x, gport->pcb_y, true, true);
           Settings.ShowSolderSide = !Settings.ShowSolderSide; /* Swapped back below */
           break;
         default:
